ion/integration_test.go

```diff
@@ -395,7 +395,7 @@ func testTextRoundTrip(t *testing.T, fp string) {
 	}
 }
 
-// Create a TextWriter from data parameter. Return the writer and string builder containing writer's contents
+// Create a TextWriter from data parameter and return the string builder containing writer's contents.
 func encodeAsTextIon(t *testing.T, data string) strings.Builder {
```

Comment:
Golang allows you to use the `string` data type to mean "arbitrary bytes", but I don't think we should use it that way. When I first read this function I was surprised because I thought if `data` was a `string`, it must already be text Ion so there'd be no need to encode it as text Ion. Only when I went and looked at the callsite did I realize that we were taking a binary buffer and calling `String()` on it to get a `string` instance.

We should make sure we're only using `string` to refer to text throughout the APIs. If this is a quick change, you can include it in the PR. Otherwise, please open an issue so we don't forget to address it.

Reply:
Fixed it in this test, create #48 to check throughout the project.